TheMovieDb/Modules/Home/HomeBuilder.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  HomeBuilder.swift
+//  TheMovieDb
+//
+//  Created by Javier Cueto on 18/11/21.
+//
+
+import UIKit.UIViewController
+
+enum HomeBuilder: HomeBuilderProtocol {
+    static func createModule() -> UIViewController {
+        let view = HomeView()
+        let service = APIService()
+        let presenter: HomePresenterProtocol & HomeInteractorOutputProtocol = HomePresenter()
+        let interactor: HomeInteractorInputProtocol & HomeRemoteDataManagerOutputProtocol = HomeInteractor()
```

Comment:
This is a personal opinion, and it depends on how you see the perspective related to SingleResponsability pattern but having more than one interface to handle this could be not related to the presenter responsability, should this be a single responsability per element or is it ok to be handle both by the same presenter, if that is the case maybe create a typealias with a more meaningful name in order to don't have a big name convention like this one :)

Reply:
I know that Presenter has two responsibilities , maybe I can fix if I use Vip pattern instead, but In this case I will use type alias.